### PR TITLE
Completed

### PR DESCRIPTION
**Pull request by bkomoff**
Bryan Komoff
46986970

### CODE REVIEW

File: ex01.ex [hunk 1]
```diff
@@ -60,7 +60,7 @@ defmodule Ex01 do
   # no explicit + operators in your function                          #
   #####################################################################
 
-  sum3a = your_anonymous_function(1, 2, 3)
+  sum3a = fn (a,b,c) -> sum2a.(sum2a.(a,b), c ) end
```
Comment (pragdave) [inline ex01.ex:63]: Slightly strange spacing here.

File: ex02.ex [hunk 2]
```diff
@@ -53,7 +53,7 @@ defmodule Ex02 do
   # if the first two elements of a list are equal                #
   ################################################################
 
-  first2equal = your_anonymous_function([])
+  first2equal = fn [a,b,_c,_d,_e] -> a == b end
```
Comment (pragdave) [inline ex02.ex:56]: ## -4

This code only works for list containing five elements. Although this passes the test, the requirement is clearly more general.

File: ex03.ex [hunk 3]
```diff
@@ -55,7 +55,19 @@ defmodule Ex03 do
 
   """
 
-  def odd_even . . . "your code"
+def odd_even([]), do: []
```
Comment (pragdave) [inline ex03.ex:58]: # -4

This only works for numbers between 1 and 9.

I was looking for something like:

~~~ elixir
  def odd_even([]), do: []
  def odd_even([ h | t ]), do: [ atom_for(h) | odd_even(t) ]

  defp atom_for(n) when Integer.is_odd(n),  do:  :odd
  defp atom_for(n) when Integer.is_even(n), do:  :even
~~~

File: ex03.ex [hunk 4]
```diff
@@ -77,8 +89,10 @@ defmodule Ex03 do
 
   """
 
-  def list_contains . .. "your code"
-
+def list_contains([], n ), do: []
```
Comment (pragdave) [inline ex03.ex:92]: # -4

The instructions say not to use any library functions.

File: ex03.ex [hunk 5]
```diff
@@ -101,9 +115,11 @@ defmodule Ex03 do
 
   """
 
-  def list_equal . . . "your code"
-
-
+def list_equal([],[]), do: []
```
Comment (pragdave) [inline ex03.ex:118]: # -5

This just checks the lengths.

Why do 

~~~
def list_equal([h1 | t1], [h2 | t2]) do
   check_length(1 + length(t1), 1 + length(t2)) 
end
~~~

and not

~~~
def list_equal(list1, list2) do
   check_length(length(list1), length(list2)) 
end
~~~

(Of course, `length/1` is a library function, which you can't use.

I was looking for something like: 

~~~ elixir
  def list_equal([], []),                              do: true
  def list_equal([ h | t1 ], [h | t2]),                do: list_equal(t1, t2)
  def list_equal(a, b) when is_list(a) and is_list(b), do: false
~~~

File: ex03.ex [hunk 6]
```diff
@@ -101,9 +115,11 @@ defmodule Ex03 do
...
+def list_equal([h1 | t1], [h2 | t2]) do
+   check_length(1 + length(t1), 1 + length(t2)) 
+end
+def check_length( a, b ), do: a == b
 
   ##############################################################################
   # 3.4:  5 points #
```
Comment (pragdave) [inline ex03.ex:125]: # 0

File: ex04.ex [hunk 7]
```diff
@@ -75,8 +79,8 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
-
+  #def even_odd([]) do: []
```
Comment (pragdave) [inline ex04.ex:82]: # 0

**Comment by pragdave** [comment]
If you'd like to meet in video chat to talk about this, just let me know.

**Comment by bkomoff** [comment]
I apologize for the poor performance. I am still recovering from the past two weeks with hurricane preparation and being without power for several days last week. I am catching up on the classes and I will run through the assignment again to make sure I understand all the concepts. I will reach out to you if I am struggling with any topics. Thank you

**Pull request closed without merging**